snakefmt/formatter.py

```diff
@@ -74,10 +75,21 @@ def read_black_config(self, path: PathLike) -> black.FileMode:
         if "line_length" not in config:
             config["line_length"] = self.line_length
 
-        try:
-            return black.FileMode(**config)
-        except TypeError as error:
-            raise InvalidBlackConfiguration(error)
+        snakecase_config = {}
+        for key, val in config.items():
+            # this is due to FileMode param being string_normalise, but CLI being
+            # skip_string_normalise - https://github.com/snakemake/snakefmt/issues/73
+            if key.startswith("skip"):
+                key = key[5:]
+                val = not val
+
+            key = key.replace("-", "_")
+            if key not in valid_black_filemode_params:
+                continue
+
+            snakecase_config[key] = val
+
+        return black.FileMode(**snakecase_config)
```

Comment:
How about only checking for `skip_string_normalization` in the config object, and if it is there, remove it and replace by `string_normalization`, with opposite value like you have done?

```
passed_config = config
if "skip_string_normalization" in config:
    string_normalize = not passed_config.pop("skip_string_normalization")
    passed_config["string_normalization"] = string_normalize
return black.FileMode(**passed_config)
```
I would hope this allows us to avoid using `valid_black_filemode_params` which should not appear in `snakefmt` at all

Also why do we need to support `skip_string_normalization`? Should user not use `string_normalization` in their config file?

Reply:
If you really want to keep your solution (I think the advantage is no longer raising `InvalidBlackConfiguration` exceptions looking at tests below), we should introspect the `FileMode` object, ie derive `valid_black_filemode_params` as the attributes of `FileMode`

Reply:
One way of achieving this would be 
```
valid_black_filemode_params = set(black.FileMode().__dict__.keys())
```
:-)

Reply:
But I personally prefer just passing the config object and raising errors as if the FileMode object changes this introspection might fail

Reply:
I'll think of a better way to handle this. The issue is not all valid `black` config params are accepted as args to `black.FileMode` as they parse these out at CLI time, whereas we are trying to parse them later on. Checking the valid params similar to what you suggest though may solve the issue.